tests/integration/conftest.py

```diff
@@ -22,47 +22,78 @@
 import pytest
 
 import libnmstate
+from libnmstate.schema import Interface
 from libnmstate.schema import InterfaceIPv6
 
 from .testlib import statelib
+from .testlib.cmd import exec_cmd
+from .testlib.env import TEST_NIC1
+from .testlib.env import TEST_NIC2
 from .testlib.statelib import INTERFACES
 
 
 @pytest.fixture(scope='session', autouse=True)
-def logging_setup():
+def test_env_setup():
     logging.basicConfig(
         format='%(asctime)s %(name)-12s %(levelname)-8s %(message)s',
         level=logging.DEBUG,
     )
+    old_state = libnmstate.show()
```

Comment:
Each veth fits under a contextmanager.
```
with net_config_env: # Not sure why we need to preserve/protect this.
    with veth1(TEST_NIC1), veth2(TEST_NIC1):
        yield
```

Reply:
Ahh I see now that the "preserve" was already there, not sure why though.
Anyway, it should probably be a contextmanager now.